OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -21,7 +21,19 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Attach this to a unit to enable dynamic upgrades by warheads, experience, crates, support powers, etc.")]
 	public class UpgradeManagerInfo : ITraitInfo, Requires<IUpgradableInfo>
 	{
-		public object Create(ActorInitializer init) { return new UpgradeManager(init); }
+		[UpgradeGrantedReference]
+		[Desc("Upgrades that can be granted/revoked from Lua scripts.")]
+		public readonly HashSet<string> Scriptable = new HashSet<string>();
+
+		[UpgradeUsedReference]
+		[Desc("Upgrades not to check whether used.")]
+		public readonly HashSet<string> IgnoredAsUnused = new HashSet<string>();
```

Comment:
So it's "add a nasty hack that goes against the point of the lint check so we don't have to define veterancy upgrades per actor type".
-1 to that...